frontend_tests/puppeteer_tests/07-navigation.ts

```diff
@@ -34,6 +34,12 @@ async function navigate_to_settings(page: Page): Promise<void> {
     await page.waitForSelector("#settings_page", {visible: true});
 
     await page.click("#settings_page .content-wrapper .exit");
+    // Wait until the overlay is completely closed.
+    await page.waitForFunction(
```

Comment:
I've used `page.waitForFunction` instead of `page.waitForSelector("selector", {visible: false})` because when I ran the puppeteer tests 100x, it seems not to wait until the overlay is completely closed.

Reply:
The right replacement is `waitForSelector` with `{hidden: true}`, not `{visible: false}`.